### PR TITLE
Add unit test cases for mech driver.

### PR DESCRIPTION
**Pull request by koteswar16**
adding unit test cases for mechanism_hp

### CODE REVIEW

File: baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+            mock.patch.object(np_drv.HPNetworkProvisioningDriver,
+                              'create_port',
+                              return_value=None)
+        ) as (is_port, cons_port, c_port):
+            self.driver.create_port_precommit(fake_context)
+            is_port.assert_called_with(fake_context)
+            cons_port.assert_called_with(fake_context)
+            c_port.assert_called_with(fake_port_dict)
+
+    '''def test_delete_port_precommit(self):
```
Comment (selvakumars2) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: please remove commented function

File: baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                      'id': 123456,
+                      'is_lag': False}}
+        return port_dict
+
+    def test_create_port_precommit(self):
+        """Test create_port_precommit method."""
+        fake_port_dict = mock.Mock()
+        fake_context = mock.Mock()
+        with contextlib.nested(
+            mock.patch.object(hp_mech.HPMechanismDriver,
```
Comment (selvakumars2) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: why we are mocking _is_port_of_interest() is being mocked even though , you are passing valid port context.

File: baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                              '_is_port_of_interest',
+                              return_value=True),
+            mock.patch.object(hp_mech.HPMechanismDriver,
+                              '_construct_port',
```
Comment (selvakumars2) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: same as above comment

File: baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    def current(self):
+        return self._port
+
+    @property
+    def original(self):
+        return self._original_port
+
+    @property
+    def network(self):
+        return self._network_context
```
Comment (selvakumars2) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: how about bind_port test cases?

Comment (selvakumars2) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: delete_port_precommit test cases could be added

File: baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+                              '_is_port_of_interest',
+                              return_value=True),
+            mock.patch.object(hp_mech.HPMechanismDriver,
+                              '_construct_port',
+                              return_value=fake_port_dict),
+            mock.patch.object(np_drv.HPNetworkProvisioningDriver,
+                              'create_port',
+                              return_value=None)
+        ) as (is_port, cons_port, c_port):
+            self.driver.create_port_precommit(fake_context)
```
Comment (selvakumars2) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: how about exception cases ? when back-end driver throws exception just ensure MechanismDriver error is thrown

Comment (selvakumars2) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: above comment is applicable for the all back-end api

File: baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py (hunk 1 above)
Comment (koteswar16) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: portbindings is not having VNIC_BAREMETAL.
Commenting this test case for now.

File: baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py (hunk 2 above)
Comment (koteswar16) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: I am passing fake_port_dict and fake_context. Not valid one.

File: baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py (hunk 4 above)
Comment (koteswar16) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: test_delete_port_precommit test case is there but in commented state. once the portbindings is updated with VNIC_TYPE, I will uncomment it.

Comment (koteswar16) [inline baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py]: no mech driver have UT for bind_port. We need to fake entire context to unit test bind_port.

**Comment by koteswar16** [comment]
updated review comments.

**Pull request merged**